Plugins/Onnx/include/Acts/Plugins/Onnx/OnnxRuntimeBase.hpp

```diff
@@ -0,0 +1,47 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <vector>
+#include <core/session/onnxruntime_cxx_api.h>
+
+namespace Acts {
+
+// General class that sets up the ONNX runtime framework for loading a ML model
+// and using it for inference.
+class OnnxRuntimeBase {
+ public:
+  /// @brief Default constructor
+  OnnxRuntimeBase() = default;
+
+  /// @brief Parametrized constructor
+  ///
+  /// @param env the ONNX runtime environment
+  /// @param modelPath the path to the ML model in *.onnx format
+  OnnxRuntimeBase(Ort::Env& env, const char* modelPath);
+
+  /// @brief Default destructor
+  ~OnnxRuntimeBase() = default;
+
+  /// @brief Run the ONNX inference function
+  ///
+  /// @param inputTensorValues The input feature values used for prediction
+  ///
+  /// @return Pointer to output values
+  float* runONNXInference(std::vector<float>& inputTensorValues) const;
```

Comment:
Where does the returned pointer point to? Are the values manually allocated by Onnx? Who is responsible for deallocating them? Does the returned values have to be modifiable or could this be `const float*` instead?

If I understand the implementation correctly, this does not have to be a `public` method and can be switched to `protected` to avoid having implementation details publi.